posthog/hogql/timings.py

```diff
@@ -0,0 +1,41 @@
+from dataclasses import dataclass, field
+from time import perf_counter
+from typing import Dict, List
+from contextlib import contextmanager
+
+from posthog.schema import QueryTiming
+
+
+@dataclass
+class HogQLTimings:
+    # Completed time in seconds for different parts of the HogQL query
+    timings: Dict[str, float] = field(default_factory=dict)
+
+    # Used for housekeeping
+    _timing_starts: Dict[str, float] = field(default_factory=dict)
+    _timing_pointer: str = "."
+
+    def __post_init__(self):
+        self._timing_starts["."] = perf_counter()
+
+    @contextmanager
+    def measure(self, key: str):
```

Comment:
Just a thought, doesn't need to be for this PR but this could also be a place where we start a sentry span, e.g. wrap the `yield` with a 
`sentry_sdk.start_span(description=full_key):`
which means these names would show up in sentry traces and crashes. Not super useful right now but could be a nice addition in the future (see https://docs.sentry.io/platforms/python/performance/instrumentation/custom-instrumentation/#add-spans-to-a-transaction )

Reply:
I added `with start_span(op=key) as span:` around the `yield`, and later send the elapsed seconds as `span.set_tag("duration", duration)`.

Reply:
Nice, sounds good